KomodoPlatform/komodo-wallet-desktop
# Zcoin activation date

ref: https://github.com/KomodoPlatform/komodo-defi-framework/pull/1922
closes: https://github.com/KomodoPlatform/komodo-wallet-desktop/issues/2354

This PR introduces the `sync_params` feature in the newly released 1.0.7 API. This allows the user to specify which block to begin sync from - previously it was hardcoded as block 1900000 (for ARRR).

![image](https://github.com/KomodoPlatform/komodo-wallet-desktop/assets/35845239/6dd5677f-aef2-40e0-b679-d4a681096c93)


After this PR is merged it will use the date selected by the user in settings, and where the setting does not exist (e.g. for zombie) it will use 2 days in the past


To test:
- From a clean wallet, activate ARRR.
- Confirm progress % is sane. 
- Select an earlier block from the settings, then restart app and try to activate again. The sync block value for the chosen date should appear in logs like `"first_sync_block":{"actual":1500000,"is_pre_sapling":false,"requested":1500000}`. You dont need to wait for full sync (it takes a long time), just confirm in logs.
- From an older wallet, repeat the above steps.

Note: Balance from transactions prior to the sync block will not be visible / included. Some sections are very, very slow.
